pallets/pallet-bonded-coins/Cargo.toml

```diff
@@ -47,13 +46,24 @@ runtime-benchmarks = [
 	"pallet-assets/runtime-benchmarks",
 	"pallet-balances/runtime-benchmarks",
 ]
+
 std = [
 	"frame-benchmarking/std",
```

Comment:
Good point. Updated in: [3073c4d](https://github.com/KILTprotocol/kilt-node/pull/789/commits/3073c4daff258801be33d0438419936247b80a39)

I also checked the other pallets. What I see is that the runtime-benchmarks is not activated for frame-benchmarking: 

```
runtime-benchmarks = [
  "frame-benchmarking",
```

Is this a bug in the other pallets? I would have assumed, the feature is important. On substrate it is normally activated.

Reply:
Apparently we also need to specify that with `dep:frame-benchmarking` to avoid creating a new feature for each optional dependency.

Reply:
> But it should be done in a different PR
yes thats for sure


> Apparently we also need to specify that with dep:frame-benchmarking to avoid creating a new feature for each optional dependency.

what do you mean exactly?

Reply:
As far as I understand from the cargo docs, this pallet should be compilable with the following command `cargo build --features frame-benchmarking`. That's because by default any optional dependency is also exposed as a feature that, when enabled, enabled the dependency. To avoid that, the dependency must be specified as `dep:<dependency>` in any of the existing features to block this behaviour. I haven't yet tested this myself, but it seems to be the case according to the docs.